### DISCUSSION TITLE
Support closing Menu on scroll

### DISCUSSION THREAD

**Discussion by michaelaflores** [chakra-ui/chakra-ui#8899] [Ideas]
### Description

I think there are cases where it's helpful, and good UX, to close a Menu on scroll. 

### Problem Statement/Justification

In lists that aren't virtualized, there's currently not a good way to prevent menus from appearing outside of the scroll container. In this example screenshot, this list item is scrolled out of view in its container, but the menu still appears. I expected `boundary` might help with this but it doesn't affect this behavior. 
<img width="373" alt="CleanShot 2023-04-11 at 11 12 20@2x" src="https://user-images.githubusercontent.com/11711706/231209553-8dfb40a2-91ab-4a4b-b187-52fcfdd1f375.png">

### Proposed Solution or API

A possibly simple solve would be allowing a prop `closeOnScroll` that closes the menu when scroll is detected.

### Alternatives

* Virtualizing the list to remove the element when scrolled out of view. However this would still result in some temporary area of scroll where the element hasn't been recycled and thus the menu is still visible.

**Comment by Zalart** [comment]
It would be great to know, if any solution was found?
The only workaround I found so far is to use css to conditionally set `visibility: hidden; pointer-events: none;` for menu list div, to utulize [popper.js hide modifier](https://popper.js.org/docs/v2/modifiers/hide/), which is active by default.
But it is not a closing menu, just hiding it to prevent visual overflows
